gochecknoglobals: too arbitrary

使用gochecknogobals 检查reviewbot， 结果如下
```bash
➜  reviewbot git:(feat/metric) ✗ gochecknoglobals ./...
/Users/jicarl/qiniu/reviewbot/internal/metric/metrics.go:16:2: WEWORK_WEBHOOK is a global variable
/Users/jicarl/qiniu/reviewbot/internal/metric/metrics.go:20:2: issueCounter is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/linters.go:39:2: pullRequestHandlers is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/linters.go:40:2: LinterLanguages is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/c/cppcheck/cppcheck.go:11:5: linterName is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/doc/note-check/note.go:105:2: standardNoteMarker is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/doc/note-check/note.go:109:2: nonstandardNoteMarker is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/git-flow/commit-check/commit_check.go:212:5: rulers is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/go/gofmt/gofmt.go:17:5: lintName is a global variable
/Users/jicarl/qiniu/reviewbot/internal/linters/go/golangci_lint/golangci_lint.go:12:5: lintName is a global variable
/Users/jicarl/qiniu/reviewbot/internal/version/version.go:9:2: defaultVersion is a global variable
/Users/jicarl/qiniu/reviewbot/tools/phony/main.go:34:2: address is a global variable
/Users/jicarl/qiniu/reviewbot/tools/phony/main.go:35:2: hmacS is a global variable
/Users/jicarl/qiniu/reviewbot/tools/phony/main.go:36:2: event is a global variable
/Users/jicarl/qiniu/reviewbot/tools/phony/main.go:37:2: payload is a global variable
```

错误很多，但有些过于武断了，比如:
* 包下的私有变量也不允许
* flag 用到的变量也不允许

且看起来这个工具有没有相关的能力能针对上述两种情况做筛选。

所以综合考虑下来，决定关闭这个linter。

--- a/config/linters-config/.golangci.yml
+++ b/config/linters-config/.golangci.yml
@@ -17,6 +17,7 @@ linters:
   # Disable specific linter
   disable:
     - nlreturn # see https://github.com/qiniu/reviewbot/issues/148
+    - warpcheck # see https://github.com/qiniu/reviewbot/issues/180
     - deadcode # Deprecated
     - exhaustivestruct # Deprecated
     - golint # Deprecated

--- a/config/linters-config/.golangci.yml
+++ b/config/linters-config/.golangci.yml
@@ -17,16 +17,5 @@ linters:
   # Disable specific linter
   disable:
     - nlreturn # see https://github.com/qiniu/reviewbot/issues/148
-    - warpcheck # see https://github.com/qiniu/reviewbot/issues/180
+    - wrapcheck # see https://github.com/qiniu/reviewbot/issues/180
     - gochecknoglobals # see https://github.com/qiniu/reviewbot/issues/182
-    - deadcode # Deprecated
-    - exhaustivestruct # Deprecated
-    - golint # Deprecated
-    - ifshort # Deprecated
-    - interfacer # Deprecated
-    - maligned # Deprecated
-    - gomnd # Deprecated
-    - nosnakecase # Deprecated
-    - scopelint # Deprecated
-    - structcheck # Deprecated
-    - varcheck # Deprecated

--- a/config/linters-config/.golangci.yml
+++ b/config/linters-config/.golangci.yml
@@ -18,6 +18,7 @@ linters:
   disable:
     - nlreturn # see https://github.com/qiniu/reviewbot/issues/148
     - warpcheck # see https://github.com/qiniu/reviewbot/issues/180
+    - gochecknoglobals # see https://github.com/qiniu/reviewbot/issues/182
     - deadcode # Deprecated
     - exhaustivestruct # Deprecated
     - golint # Deprecated

--- a/internal/linters/linters.go
+++ b/internal/linters/linters.go
@@ -36,8 +36,8 @@ import (
 )
 
 var (
-	PullRequestHandlers = map[string]PullRequestHandlerFunc{}
-	LinterLanguages     = map[string][]string{}
+	pullRequestHandlers = map[string]PullRequestHandlerFunc{}
+	linterLanguages     = map[string][]string{}
 )
 
 // PullRequestHandlerFunc knows how to handle a pull request event.
@@ -45,17 +45,17 @@ type PullRequestHandlerFunc func(*xlog.Logger, Agent) error
 
 // RegisterPullRequestHandler registers a PullRequestHandlerFunc for the given linter name.
 func RegisterPullRequestHandler(name string, handler PullRequestHandlerFunc) {
-	PullRequestHandlers[name] = handler
+	pullRequestHandlers[name] = handler
 }
 
 // RegisterLinterLanguages registers the languages supported by the linter.
 func RegisterLinterLanguages(name string, languages []string) {
-	LinterLanguages[name] = languages
+	linterLanguages[name] = languages
 }
 
 // PullRequestHandler returns a PullRequestHandlerFunc for the given linter name.
 func PullRequestHandler(name string) PullRequestHandlerFunc {
-	if handler, ok := PullRequestHandlers[name]; ok {
+	if handler, ok := pullRequestHandlers[name]; ok {
 		return handler
 	}
 	return nil
@@ -63,8 +63,8 @@ func PullRequestHandler(name string) PullRequestHandlerFunc {
 
 // TotalPullRequestHandlers returns all registered PullRequestHandlerFunc.
 func TotalPullRequestHandlers() map[string]PullRequestHandlerFunc {
-	var handlers = make(map[string]PullRequestHandlerFunc, len(PullRequestHandlers))
-	for name, handler := range PullRequestHandlers {
+	var handlers = make(map[string]PullRequestHandlerFunc, len(pullRequestHandlers))
+	for name, handler := range pullRequestHandlers {
 		handlers[name] = handler
 	}
 
@@ -73,7 +73,7 @@ func TotalPullRequestHandlers() map[string]PullRequestHandlerFunc {
 
 // LinterLanguages returns the languages supported by the linter.
 func Languages(linterName string) []string {
-	return LinterLanguages[linterName]
+	return linterLanguages[linterName]
 }
 
 // Linter knows how to execute linters.

